client/src/pages/Feed.js

```diff
@@ -586,8 +597,19 @@ const Feed = (props) => {
   }, [applyFilters, selectedOptions, location]); // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
+    if (typeof queryParams.s_keyword !== "undefined") {
+      setSortValue("relevance");
+    } else if (!ignoreUserLocation) {
+      setSortValue("proximity");
+    } else {
+      setSortValue(t("feed.filters.sortBy"));
+    }
     refetchPosts(); // will trigger loadPosts(if needed) (by toggling toggleRefetch)
-  }, [queryParams]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [queryParams, ignoreUserLocation]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
What is this eslint-disable... for?

Reply:
for some reason the linter wants to add anything that gets mentioned in the useEffect to the dependency array. So without the eslint-disable on commit sortValue gets added to the dependency array. Since we are using setSortValue in that useEffect it changes the state but then runs again, creating an infinite loop